src/components/cursor-controller.js

```diff
@@ -2,451 +2,190 @@ const TARGET_TYPE_NONE = 1;
-    this._handleCursorLoaded = this._handleCursorLoaded.bind(this);
-    this._handleControllerConnected = this._handleControllerConnected.bind(this);
-    this._handleControllerDisconnected = this._handleControllerDisconnected.bind(this);
+    this.forceCursorUpdate = this.forceCursorUpdate.bind(this);
+    this.startInteraction = this.startInteraction.bind(this);
+    this.moveCursor = this.moveCursor.bind(this);
+    this.endInteraction = this.endInteraction.bind(this);
+    this.changeDistanceMod = this.changeDistanceMod.bind(this);
+    this.setRaycasterWithMousePos = this.setRaycasterWithMousePos.bind(this);
+    this.updateDistanceAndTargetType = this.updateDistanceAndTargetType.bind(this);
 
+    this._handleCursorLoaded = this._handleCursorLoaded.bind(this);
     this.data.cursor.addEventListener("loaded", this._handleCursorLoaded);
   },
 
-  remove: function() {
-    this.data.cursor.removeEventListener("loaded", this._handleCursorLoaded);
+  enable: function() {
+    this.enabled = true;
   },
 
-  update: function(oldData) {
-    if (oldData.physicalHandSelector !== this.data.physicalHandSelector) {
-      this._handleModelLoaded();
-    }
-
-    if (oldData.handedness !== this.data.handedness) {
-      //TODO
-    }
+  disable: function() {
+    this.enabled = false;
+    this.setCursorVisibility(false);
   },
 
-  play: function() {
-    document.addEventListener("touchstart", this._handleTouchStart);
-    document.addEventListener("touchmove", this._handleTouchMove);
-    document.addEventListener("touchend", this._handleTouchEnd);
-    document.addEventListener("touchcancel", this._handleTouchEnd);
-    document.addEventListener("mousedown", this._handleMouseDown);
-    document.addEventListener("mousemove", this._handleMouseMove);
-    document.addEventListener("mouseup", this._handleMouseUp);
-    document.addEventListener("wheel", this._handleWheel);
-
-    window.addEventListener("enter-vr", this._handleEnterVR);
-    window.addEventListener("exit-vr", this._handleExitVR);
-
-    this.data.playerRig.addEventListener(this.data.primaryDown, this._handlePrimaryDown);
-    this.data.playerRig.addEventListener(this.data.primaryUp, this._handlePrimaryUp);
-    this.data.playerRig.addEventListener(this.data.grabEvent, this._handlePrimaryDown);
-    this.data.playerRig.addEventListener(this.data.releaseEvent, this._handlePrimaryUp);
-    this.data.playerRig.addEventListener("cardboardbuttondown", this._handlePrimaryDown);
-    this.data.playerRig.addEventListener("cardboardbuttonup", this._handlePrimaryUp);
-
-    this.data.playerRig.addEventListener("model-loaded", this._handleModelLoaded);
-
-    this.el.sceneEl.addEventListener("controllerconnected", this._handleControllerConnected);
-    this.el.sceneEl.addEventListener("controllerdisconnected", this._handleControllerDisconnected);
+  update: function() {
+    if (this.data.rayObject) {
+      this.rayObject = this.data.rayObject.object3D;
```

Comment:
This is odd, why the extra refference? If you are just doing it to save typing, I would make a local variable in tick.

Reply:
👍